Components/Queues/Handler/PhpAmqpHandler.php

```diff
@@ -0,0 +1,230 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Smartbox\Integration\FrameworkBundle\Components\Queues\Handler;
+
+use JMS\Serializer\SerializerBuilder;
+use JMS\Serializer\SerializerInterface;
+use mysql_xdevapi\Exception;
+use PhpAmqpLib\Channel\AMQPChannel;
+use PhpAmqpLib\Message\AMQPMessage;
+use Psr\Log\LoggerAwareInterface;
+use Psr\Log\LoggerAwareTrait;
+use Smartbox\CoreBundle\Type\SerializableInterface;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\QueueMessage;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\QueueMessageHandlerInterface;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\QueueMessageInterface;
+use Smartbox\Integration\FrameworkBundle\Core\Consumers\IsStopableConsumer;
+use Smartbox\Integration\FrameworkBundle\Core\Endpoints\EndpointFactory;
+use Smartbox\Integration\FrameworkBundle\Core\Endpoints\EndpointInterface;
+use Smartbox\Integration\FrameworkBundle\DependencyInjection\Traits\UsesSmartesbHelper;
+use Smartbox\Integration\FrameworkBundle\Exceptions\Handler\UsesExceptionHandlerTrait;
+
+/**
+ * Class PhpAmqpHandler
+ * @package Smartbox\Integration\FrameworkBundle\Components\Queues\Handler
+ */
+class PhpAmqpHandler implements LoggerAwareInterface
+{
+    use LoggerAwareTrait;
+    use UsesSmartesbHelper;
+    use UsesExceptionHandlerTrait;
+    use IsStopableConsumer;
```

Comment:
Which is the reason of this handler has the responsibility of stopping the consumer?